//
// DO NOT MODIFY THIS FILE DIRECTLY!
// ITS IS GENERATED BY generate_policy_source.py
// FROM policy_templates.json
//

syntax = "proto2";

option optimize_for = LITE_RUNTIME;

package enterprise_management;

// PBs for individual settings.

message PolicyOptions {
  enum PolicyMode {
    // The user may choose to override the given settings.
    RECOMMENDED = 1;
    // The given settings are applied regardless of user choice.
    MANDATORY = 2;
  }
  optional PolicyMode mode = 1;
}

message HomepageProto {
  optional PolicyOptions policy_options = 1;
  optional string HomepageLocation = 2;
  optional bool HomepageIsNewTabPage = 3;
}

message ApplicationLocaleValueProto {
  optional PolicyOptions policy_options = 1;
  optional string ApplicationLocaleValue = 2;
}

message AlternateErrorPagesEnabledProto {
  optional PolicyOptions policy_options = 1;
  optional bool AlternateErrorPagesEnabled = 2;
}

message SearchSuggestEnabledProto {
  optional PolicyOptions policy_options = 1;
  optional bool SearchSuggestEnabled = 2;
}

message DnsPrefetchingEnabledProto {
  optional PolicyOptions policy_options = 1;
  optional bool DnsPrefetchingEnabled = 2;
}

message DisableSpdyProto {
  optional PolicyOptions policy_options = 1;
  optional bool DisableSpdy = 2;
}

message JavascriptEnabledProto {
  optional PolicyOptions policy_options = 1;
  optional bool JavascriptEnabled = 2;
}

message SavingBrowserHistoryDisabledProto {
  optional PolicyOptions policy_options = 1;
  optional bool SavingBrowserHistoryDisabled = 2;
}

message PrintingEnabledProto {
  optional PolicyOptions policy_options = 1;
  optional bool PrintingEnabled = 2;
}

message SafeBrowsingEnabledProto {
  optional PolicyOptions policy_options = 1;
  optional bool SafeBrowsingEnabled = 2;
}

message MetricsReportingEnabledProto {
  optional PolicyOptions policy_options = 1;
  optional bool MetricsReportingEnabled = 2;
}

message PasswordManagerProto {
  optional PolicyOptions policy_options = 1;
  optional bool PasswordManagerEnabled = 2;
  optional bool PasswordManagerAllowShowPasswords = 3;
}

message AutoFillEnabledProto {
  optional PolicyOptions policy_options = 1;
  optional bool AutoFillEnabled = 2;
}

message DisabledPluginsProto {
  optional PolicyOptions policy_options = 1;
  repeated string DisabledPlugins = 2;
}

message SyncDisabledProto {
  optional PolicyOptions policy_options = 1;
  optional bool SyncDisabled = 2;
}

message ProxyProto {
  optional PolicyOptions policy_options = 1;
  optional string ProxyMode = 2;
  optional int64 ProxyServerMode = 3;
  optional string ProxyServer = 4;
  optional string ProxyPacUrl = 5;
  optional string ProxyBypassList = 6;
}

message HTTPAuthenticationProto {
  optional PolicyOptions policy_options = 1;
  optional string AuthSchemes = 2;
  optional bool DisableAuthNegotiateCnameLookup = 3;
  optional bool EnableAuthNegotiatePort = 4;
  optional string AuthServerWhitelist = 5;
  optional string AuthNegotiateDelegateWhitelist = 6;
  optional string GSSAPILibraryName = 7;
}

message ExtensionsProto {
  optional PolicyOptions policy_options = 1;
  repeated string ExtensionInstallBlacklist = 2;
  repeated string ExtensionInstallWhitelist = 3;
  repeated string ExtensionInstallForcelist = 4;
}

message ShowHomeButtonProto {
  optional PolicyOptions policy_options = 1;
  optional bool ShowHomeButton = 2;
}

message DeveloperToolsDisabledProto {
  optional PolicyOptions policy_options = 1;
  optional bool DeveloperToolsDisabled = 2;
}

message RestoreOnStartupGroupProto {
  optional PolicyOptions policy_options = 1;
  optional int64 RestoreOnStartup = 2;
  repeated string RestoreOnStartupURLs = 3;
}

message DefaultSearchProviderProto {
  optional PolicyOptions policy_options = 1;
  optional bool DefaultSearchProviderEnabled = 2;
  optional string DefaultSearchProviderName = 3;
  optional string DefaultSearchProviderKeyword = 4;
  optional string DefaultSearchProviderSearchURL = 5;
  optional string DefaultSearchProviderSuggestURL = 6;
  optional string DefaultSearchProviderInstantURL = 7;
  optional string DefaultSearchProviderIconURL = 8;
  repeated string DefaultSearchProviderEncodings = 9;
}

message ContentSettingsProto {
  optional PolicyOptions policy_options = 1;
  optional int64 DefaultCookiesSetting = 2;
  optional int64 DefaultImagesSetting = 3;
  optional int64 DefaultJavaScriptSetting = 4;
  optional int64 DefaultPluginsSetting = 5;
  optional int64 DefaultPopupsSetting = 6;
  optional int64 DefaultNotificationSetting = 7;
  optional int64 DefaultGeolocationSetting = 8;
}

message Disable3DAPIsProto {
  optional PolicyOptions policy_options = 1;
  optional bool Disable3DAPIs = 2;
}

message ChromeFrameRendererSettingsProto {
  optional PolicyOptions policy_options = 1;
  optional int64 ChromeFrameRendererSettings = 2;
  repeated string RenderInChromeFrameList = 3;
  repeated string RenderInHostList = 4;
}

message ChromeFrameContentTypesProto {
  optional PolicyOptions policy_options = 1;
  repeated string ChromeFrameContentTypes = 2;
}

message ChromeOsLockOnIdleSuspendProto {
  optional PolicyOptions policy_options = 1;
  optional bool ChromeOsLockOnIdleSuspend = 2;
}


// --------------------------------------------------
// Wrapper PB for DMServer -> ChromeOS communication.

message CloudPolicySettings {
  optional HomepageProto Homepage = 1;
  optional ApplicationLocaleValueProto ApplicationLocaleValue = 2;
  optional AlternateErrorPagesEnabledProto AlternateErrorPagesEnabled = 3;
  optional SearchSuggestEnabledProto SearchSuggestEnabled = 4;
  optional DnsPrefetchingEnabledProto DnsPrefetchingEnabled = 5;
  optional DisableSpdyProto DisableSpdy = 6;
  optional JavascriptEnabledProto JavascriptEnabled = 7;
  optional SavingBrowserHistoryDisabledProto SavingBrowserHistoryDisabled = 8;
  optional PrintingEnabledProto PrintingEnabled = 9;
  optional SafeBrowsingEnabledProto SafeBrowsingEnabled = 10;
  optional MetricsReportingEnabledProto MetricsReportingEnabled = 11;
  optional PasswordManagerProto PasswordManager = 12;
  optional AutoFillEnabledProto AutoFillEnabled = 13;
  optional DisabledPluginsProto DisabledPlugins = 14;
  optional SyncDisabledProto SyncDisabled = 15;
  optional ProxyProto Proxy = 16;
  optional HTTPAuthenticationProto HTTPAuthentication = 17;
  optional ExtensionsProto Extensions = 18;
  optional ShowHomeButtonProto ShowHomeButton = 19;
  optional DeveloperToolsDisabledProto DeveloperToolsDisabled = 20;
  optional RestoreOnStartupGroupProto RestoreOnStartupGroup = 21;
  optional DefaultSearchProviderProto DefaultSearchProvider = 22;
  optional ContentSettingsProto ContentSettings = 23;
  optional Disable3DAPIsProto Disable3DAPIs = 24;
  optional ChromeFrameRendererSettingsProto ChromeFrameRendererSettings = 25;
  optional ChromeFrameContentTypesProto ChromeFrameContentTypes = 26;
  optional ChromeOsLockOnIdleSuspendProto ChromeOsLockOnIdleSuspend = 27;
}
